scripts/runit_hvd_osprey.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+#SBATCH -N 1
+##SBATCH --ntasks-per-node=8
+##SBATCH -n 8
+####SBATCH -x "nid[000076-000080,000137-000144]"
+####SBATCH --contiguous
+#SBATCH -t 2:00:00
+#SBATCH --exclusive
+#SBATCH -p spider
+#SBATCH -C V100
+#SBATCH --job-name=cosmoflow-gpu
+#####SBATCH --nodelist=nid[000256-000287]
+#####SBATCH --nodelist=nid[000256-000383]
+###SBATCH --nodelist=nid[000256-000512]
+
+###SBATCH --core-spec=2
+ulimit -l unlimited
+source ./config_cuda10.sh
+unset PYTHONPATH
+#module rm PrgEnv-cray
+export SCRATCH=/lus/scratch/jbalma
+INSTALL_DIR=/lus/scratch/jbalma/condenv-cuda10-cosmoflow
+export CRAY_CPU_TARGET=x86-64
+#conda create -y --prefix $INSTALL_DIR python=3.6 cudatoolkit=10.0 cudnn
+source activate $INSTALL_DIR
+#conda activate $INSTALL_DIR
+export PATH=${INSTALL_DIR}/bin:${PATH} #/home/users/${USER}/.local/bin:${PATH}
+
+echo $CUDATOOLKIT_HOME
+which gcc
+which python
+BUILD_TIMEMORY=0
+
+if [ $BUILD_TIMEMORY -eq 0 ]
+then
+    echo "Running assuming you already have timemory installed"
+else
+
+#fi
+#pip install tensorflow-gpu==1.15
+#Install Horovod on Vanilla Cluster with GPU, OpenMPI support and no NCCL
+#CMAKE_CXX_COMPILER=$MPI_CXX CMAKE_CC_COMPILER=$MPI_CC CXX=mpicxx CC=mpicc HOROVOD_CUDA_HOME=${CUDATOOLKIT_HOME} HOROVOD_MPICXX_SHOW="mpicxx -show" HOROVOD_MPI_HOME=${MPI_PATH} HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITHOUT_PYTORCH=1 HOROVOD_WITHOUT_MXNET=1 pip install --global-option=build_ext --global-option="-I ${CUDATOOLKIT_HOME}/include" -v --no-cache-dir --force-reinstall horovod
+
+#Install on XC50
+#pip uninstall horovod
+#cc=cc CXX=CC HOROVOD_HIERARCHICAL_ALLREDUCE=1 HOROVOD_MPICXX_SHOW="CC --cray-print-opts=all" pip install -v --no-cache-dir horovod
+#ulimit -c
+
+#Install Horovod on Cluster without GPU support
+#CXX=mpic++ CC=gcc HOROVOD_MPICXX_SHOW="mpic++ -show" HOROVOD_MPI_HOME=${MPI_PATH} HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITHOUT_PYTORCH=1 HOROVOD_WITHOUT_MXNET=1 pip install --global-option=build_ext --global-option="-I ${CUDATOOLKIT_HOME}/include" --no-cache-dir --force-reinstall horovod
+
+# Setup a build directory
+TIMEMORY_BUILD_DIR=$SCRATCH/timemory-tf-1.13.1-py36
+rm -rf $TIMEMORY_BUILD_DIR
+mkdir -p $TIMEMORY_BUILD_DIR && cd $TIMEMORY_BUILD_DIR
+
+# Do a clean checkout
+[ -d timemory ] && rm -rf timemory
+git clone https://github.com/NERSC/timemory.git
+
+# Install dependencies missing from the TF installation
+python -m pip install scikit-build
+python -m pip install pandas
+# Configure the build
+export BUILD_SHARED_LIBS=1
+export CRAYPE_LINK_TYPE=dynamic
+
+# Run the build
+cd timemory
+python -m pip uninstall timemory
+python -m pip install -r requirements.txt --user
+#python setup.py install <ARGS> -- <CMAKE_ARGS>
+#python setup.py install --help
+python -m pip install scikit-build
+export CMAKE_CXX_COMPILER=$MPI_CXX
```

Comment:
Just a side note, this env variable isn't used by CMake. You can use either `-DCMAKE_CXX_COMPILER=$MPI_CXX` or `export CXX=$MPI_CXX`. The former will force the compiler to change if there is an existing build and the CXX environment variable is used if there is no existing build and the latter is not specified on the command line.

Also, out of curiosity, is there a particular reason you are using the MPI compiler wrappers here? The find_package for MPI in CMake should take care of that but if you've had build errors, please lmk.